FirebaseSessions/Sources/Settings/SettingsProtocol.swift

```diff
@@ -17,13 +17,15 @@ import Foundation
 
 /// Provides the APIs to access Settings and their configuration values
 protocol SettingsProtocol {
+  // Update the settings for all the settings providers
+  func updateSettings()
```

Comment:
Curious on this refactor: After the previous refactor, we had two protocols with the following needs:

1. SettingsProvider - Provides settings from any service (eg: Crashlytics settings, SDK defaults, LocalOverrides)
2. SettingsProtocol - Just a service that provides the config values.

With the new refactor, we have a new API updateSettings in the SettingsProtocol. Feel like the SettingsProtocol does more than just abstracting the configs. What do you think would be the purpose of the SettingsProtocol in this new behavior?

Reply:
The purpose of SettingsProtocol should be the same - just that it abstracts the configs. The only difference is before we were calling updateSettings on the SessionsSettings class directly instead of using the protocol internally: https://github.com/firebase/firebase-ios-sdk/blob/de6eb3f971ddb9c6ad0a27b78d698a95d1d77d3e/FirebaseSessions/Sources/FirebaseSessions.swift#L177

See https://github.com/firebase/firebase-ios-sdk/pull/10669/files#diff-04ac03aa6a3c3aae8fb90eb2b10c05ee112682853acbd903aa7abc7db089e94cR39